File: ai-solutions/01/gemini-3-pro/python/main.py
import sys

def solve():
    if len(sys.argv) < 2:
        print("Usage: python main.py <input_file>")
        sys.exit(1)
        
    input_file = sys.argv[1]
    try:
        with open(input_file, 'r') as f:
            lines = f.read().splitlines()
    except FileNotFoundError:
        print(f"Error: Input file '{input_file}' not found.")
        sys.exit(1)

    pos = 50
    p1_count = 0
    p2_count = 0

    for line in lines:
        if not line: continue
        direction = line[0]
        amount = int(line[1:])

        # Simulate click by click for Part 2
        for _ in range(amount):
            if direction == 'R':
                pos = (pos + 1) % 100
            else:
                pos = (pos - 1) % 100
            
            if pos == 0:
                p2_count += 1
        
        # Check end position for Part 1
        if pos == 0:
            p1_count += 1

    print(p1_count)
    print(p2_count)

if __name__ == "__main__":
    solve()

File: ai-solutions/01/gemini-3-pro/rust/src/main.rs
use std::env;
use std::fs;
use std::process;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        eprintln!("Usage: {} <input_file>", args[0]);
        process::exit(1);
    }

    let input_file = &args[1];
    let content = match fs::read_to_string(input_file) {
        Ok(c) => c,
        Err(e) => {
            eprintln!("Error reading file '{}': {}", input_file, e);
            process::exit(1);
        }
    };

    let mut pos = 50;
    let mut p1_count = 0;
    let mut p2_count = 0;

    for line in content.lines() {
        if line.trim().is_empty() {
            continue;
        }

        let direction = line.chars().next().unwrap();
        let amount: i32 = line[1..].parse().unwrap();

        for _ in 0..amount {
            if direction == 'R' {
                pos = (pos + 1) % 100;
            } else {
                pos = (pos - 1 + 100) % 100;
            }

            if pos == 0 {
                p2_count += 1;
            }
        }

        if pos == 0 {
            p1_count += 1;
        }
    }

    println!("{}", p1_count);
    println!("{}", p2_count);
}

